test/new-e2e/pkg/utils/optional/optional.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package optional
+
+func MakeParams[Param any, Option ~func(*Param) error](options ...Option) (*Param, error) {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
TIL: https://go.dev/ref/spec#Underlying_types